## The command line for bioinformatics workflows

In this lesson we'll go through an introduction to what sequencing is, learn about the genomic data file types and how to start processing genomic data using bioinformatics tools at the command line.

## Learning objectives

- Knowledge of cloud computing resources and why it's important in bioinformatics
- Understand how to connect to cloud computing resources via ssh
- Review navigating at the command line
- Understand the genomic data file format, FASTQ
- How to use command line functions to look at and get information about FASTQ files
- Understand a general bioinformatics workflow
- Understand how to use command line bioinformatics tools to process genomic data, using FastQC as an example
- Know where to find more information on working with genomic data and specific analyses

## Lesson Outline

### I. Using cloud computing for fun and genomic data
**Goals:** Discuss needs for cloud computing and what are some options. Connect to Amazon AWS resources using ssh. Provide documentation on how to get access to cloud computing resources.

**Materials:**  [Introduction to cloud computing](http://www.datacarpentry.org/cloud-genomics/02-why-cloud-computing/)  

**Additional resources:**  
- [Setting up instances on Amazon AWS and CyVerse for remote computing](http://www.datacarpentry.org/cloud-genomics/discuss/)
- [JetStream for remote computing](https://angus.readthedocs.io/en/2017/jetstream/boot.html)

### II. The genomic data file type and the command line
**Goals:** Work with the FASTQ file type at the command line.

**Materials:**  
- [Introduction to the command line](https://data-lessons.github.io/shell-genomics/)

### III. Running bioinformatics programs
**Goals:** How to run bioinformatics programs at the command line, using FastQC as an example.

**Materials:** [Bioinformatics programs - quality control](http://www.datacarpentry.org/wrangling-genomics/00-readQC/)


### Other Resources

#### Lessons
- [Data Carpentry Genomics workshop](https://data-lessons.github.io/genomics-workshop/)
- [EDAMAME materials](https://github.com/edamame-course/2016-tutorials/wiki/Schedule-EDAMAME-2016) - short course in microbial metagenomics
- [Angus NGS materials](https://angus.readthedocs.io/en/2017/toc.html) - short course on next generation sequencing   
- [Amplicon sequencing: mothur SOP](https://www.mothur.org/wiki/MiSeq_SOP)
- [Amplicon sequencing: mothur tutorial](https://github.com/tracykteal/mothur-tutorial)
- [Amplicon sequencing: QIIME](http://qiime.org)
- [Amplicon sequencing: QIIME tutorial](https://github.com/edamame-course/2015-tutorials/blob/master/final/2015-06-23-QIIME1.md)


#### Articles
- [Article: Computing workflows for biologists](http://journals.plos.org/plosbiology/article?id=10.1371/journal.pbio.1002303)
- [Article: Good enough practices in scientific computing](http://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1005510)
